### PR TITLE
chore(rds): add MariaDB 10.5.13, 10.4.22, 10.3.32, 10.2.41 and Aurora MySQL 3.01.0

### PR DESCRIPTION
**Pull request by jumic**
Add new RDS versions:

**MariaDbEngineVersion 10.5.13, 10.4.22, 10.3.32 and 10.2.41**
[Announcement](https://aws.amazon.com/about-aws/whats-new/2021/12/amazon-rds-mariadb-supports-new-minor-versions/)

**AuroraMysqlEngineVersion 3.01.0**
[Announcement](https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-aurora-mysql-8-0/)
Version informations retrieved from CLI command: `aws rds describe-db-engine-versions --region us-east-1 --engine aurora-mysql --engine-version 8.0`

Deployment tested successfully:
```ts
new rds.DatabaseCluster(this, 'DatabaseCluster', {
  engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_01_0 }),
  instanceProps: {
    instanceType: ec2.InstanceType.of(ec2.InstanceClass.R5, ec2.InstanceSize.XLARGE),
    vpc,
  },
  removalPolicy: RemovalPolicy.DESTROY,
});
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/cluster-engine.ts [hunk 1]
```diff
@@ -355,6 +357,10 @@ export class AuroraMysqlEngineVersion {
     return new AuroraMysqlEngineVersion(`5.7.${addStandardPrefix ? 'mysql_aurora.' : ''}${minorVersion}`);
   }
 
+  private static builtIn_8_0(minorVersion: string, addStandardPrefix: boolean = true): AuroraMysqlEngineVersion {
+    return new AuroraMysqlEngineVersion(`8.0.${addStandardPrefix ? 'mysql_aurora.' : ''}${minorVersion}`, '8.0');
+  }
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster-engine.ts:362]: I doubt we will ever use this with the second parameter as `false`, so let's just remove it completely for now:

```suggestion
  private static builtIn_8_0(minorVersion: string): AuroraMysqlEngineVersion {
    return new AuroraMysqlEngineVersion(`8.0.mysql_aurora.${minorVersion}`, '8.0');
  }
```

This is a private method, we can always add back the parameter if we need to.

File: packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts [hunk 2]
```diff
@@ -71,6 +71,20 @@ describe('cluster engine', () => {
...
+    const engine = DatabaseClusterEngine.auroraMysql({
+      version: AuroraMysqlEngineVersion.VER_3_01_0,
+    });
+
+    // WHEN
+    const family = engine.parameterGroupFamily;
+
+    // THEN
+    expect(family).toEqual('aurora-mysql8.0');
+
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster-engine.test.ts:85]: No need for this extra empty line:

```suggestion
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @jumic! A tiny change please before we merge this in 🙂.

**Comment by jumic** [comment]
Thanks for your review. I implemented the suggested changes.

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @jumic!

**Pull request merged**